python/cudf/cudf/core/column/column.py

```diff
@@ -52,6 +53,10 @@
     IntervalDtype,
     ListDtype,
     StructDtype,
+    _is_interval_dtype,
+    is_decimal_dtype,
+    is_list_dtype,
+    is_struct_dtype,
```

Comment:
IMO I'm generally not fond of using these functions since they tend to allow more objects than what is expected (like columns or uninitialized classes).

Could you make new `is_dtype_obj_*` (like we have `is_dtype_obj_numeric` in `cudf/utils/dtypes.py`) and use those?